src/app/[locale]/(main)/quiz/add/components/add-quizbank-form.tsx

```diff
@@ -74,16 +84,29 @@ const addQuizbankSchema = z.object({
       })
       .default({ question: "", answer: "" })
   ),
+  explaination: z.string({
+    required_error: "errors.invalid_type_received_undefined",
+  }).optional()
 })
 
 export type AddQuizbank = z.infer<typeof addQuizbankSchema>
 
 type AddQuizbankFormProps = {
   initialValues?: AddQuizbank
+  // action if the form
+  action?: EQuizBankAction
+  /**
+   * needed when action is edit
+   */
+  quizBankId?: string,
 }
-function AddQuizbankForm({ initialValues }: AddQuizbankFormProps) {
+function AddQuizbankForm({
+  initialValues,
+  action = EQuizBankAction.Add,
+  quizBankId
+}: AddQuizbankFormProps) {
   const errori18n = useTranslations("Validations")
-  const i18n = useTranslations("AddQuiz")
+  const i18n = useTranslations("EditQuiz")
```

Comment:
Add Quiz and Edit quiz have diff range of terms, separate this

Reply:
But separate Add Quiz and Edit Quiz means that I have to passin NextIntl like this 
-
![image](https://github.com/blackmouse572/quizler-fe/assets/81068699/c7f42f86-b5cd-42a4-85f5-9ff471b51be8)
- 
![image](https://github.com/blackmouse572/quizler-fe/assets/81068699/0e9f667f-60e6-43c5-8cff-9964a52f0f4c)

Which cause error like this 
![image](https://github.com/blackmouse572/quizler-fe/assets/81068699/986a6add-e821-4927-83ea-2b9159dde4f7)

For solution like this 
![image](https://github.com/blackmouse572/quizler-fe/assets/81068699/e952e969-02e4-4c1d-b698-69fade874d56)

I can resolve it, but that means I should create 2 form for 2 action. Which is, in my opinion, waste of code
@blackmouse572

Reply:
Seperate the form with messages props. using mode to pass correct messages

Reply:
Or using same namespace, with different type